convex/rooms.ts

```diff
@@ -0,0 +1,41 @@
+import { query, mutation } from "./_generated/server";
+import { v } from "convex/values";
+
+// create room
+export const createRoom = mutation({
+  args: { username: v.string() },
+  handler: async (ctx, args) => {
+    const roomName = args.username + "'s Room";
+    const roomId = await ctx.db.insert("rooms", { roomName });
+    const playerId = await ctx.db.insert("player", {
+      username: args.username,
+      roomId: roomId,
+    });
+    return { playId: playerId, roomId: roomId };
+  },
+});
+
+// let somebody join into room
+export const joinRoomByRoomId = mutation({
+  args: { username: v.string(), roomId: v.id("rooms") },
+  handler: async (ctx, args) => {
+    const players = await ctx.db
+      .query("player")
+      .filter((q) => q.eq(q.field("roomId"), args.roomId))
+      .collect();
+    if (players.length > 8) {
+      return {
+        message: "The backend got an error",
```

Comment:
Why message and error? We can just send one of them.

Reply:
No, here I must check the rooms all player's amount, because a room can only maximum 8 player